EpiResponsivePicture/TypeScript/src/ResponsiveImage.tsx

```diff
@@ -0,0 +1,203 @@
+  ScaleMode,
+  IFocalPoint,
+  IPictureProfile,
+  IPictureSource,
+} from "../src/generated";
+export function getResizedImageUrl(
+  imageUrl: string,
+  width: number | null = null,
+  height: number | null = null,
+  mode: ScaleMode | null = null,
+  quality: number | null = null,
+  focalPoint: IFocalPoint | null = null
+) {
+  const urlParams = [];
+
+  if (width) {
+    urlParams.push(`width=${width.toString()}`);
+  }
+
+  if (height) {
+    urlParams.push(`height=${height.toString()}`);
+  }
+
+  if (mode) {
+    const modeString = ScaleMode[mode];
+    urlParams.push(`rmode=${modeString}`);
+  }
+
+  if (quality) {
+    urlParams.push(`quality=${quality.toString()}`);
+  }
+
+  if (focalPoint) {
+    urlParams.push(`rxy=${focalPoint.x.toFixed(3)},${focalPoint.y.toFixed(3)}`);
+  }
+
+  return urlParams.length > 0 ? `${imageUrl}?${urlParams.join("&")}` : imageUrl;
+}
+
+export interface ResponsivePictureProps {
+  model: ResponsiveImageViewModel;
+  profile: IPictureProfile;
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  imgAttributes?: any;
+  className?: string;
+  fallbackAltText?: string;
+}
+
+export class ResponsivePicture extends React.Component<ResponsivePictureProps> {
+  private static buildResizedImageUrl(
+    imageUrl: string,
+    width: number,
+    mode?: ScaleMode,
+    targetAspectRatio?: IAspectRatio,
+    quality?: number,
+    image?: ResponsiveImageViewModel,
+    maxImageDimension?: number
+  ) {
+    const maxDimension = maxImageDimension || 3200;
+
+    let usedWidth = Math.min(width, maxDimension);
+    let height = null;
+
+    if (mode !== ScaleMode.Default && mode !== ScaleMode.Max && image != null) {
+      if (targetAspectRatio == null || targetAspectRatio.ratio < 0) {
+        throw "Aspect ratio is required when ScaleMode is other than Max";
+      }
+
+      height = Math.round(usedWidth / targetAspectRatio.ratio);
+      if (height > maxDimension) {
+        height = maxDimension;
+        usedWidth = height * targetAspectRatio.ratio;
+      }
+    }
+
+    return getResizedImageUrl(
+      imageUrl,
+      width,
+      height,
+      mode,
+      quality,
+      image?.focalPoint
+    );
+  }
+
+  /* eslint-disable-next-line @typescript-eslint/member-ordering */
+  constructor(props: ResponsivePictureProps) {
+    super(props);
+
+    this.createSourceElement = this.createSourceElement.bind(this);
+  }
+
+  componentDidMount() {
+    // @ts-ignore
+    if (window.objectFitPolyfill) {
+      // @ts-ignore
+      window.objectFitPolyfill();
+    }
+  }
+
+  private getDefaultUrl(): string {
+    const { sources } = this.props.profile;
+    const aspectRatio =
+      sources && sources.length > 0
+        ? sources[sources.length - 1].targetAspectRatio
+        : undefined;
+
+    if (aspectRatio) {
+      return ResponsivePicture.buildResizedImageUrl(
+        this.getImageUrl(),
+        this.props.profile.defaultWidth,
+        3,
+        aspectRatio,
+        50,
```

Comment:
I wonder if maybe it's better idea to have default quality set to 99 ?

Reply:
it is good idea to leave image untouched in terms of quality when defaults are used